File: Java/LengthOfLastWord.java
public class LengthOfLastWord {
  public static void main(String args[]){
    System.out.println("Test 1: \'Hello World\' = " + lengthOfLastWord("Hello World"));
    System.out.println("Test 2: \'   fly me   to   the moon  \' = " + lengthOfLastWord("   fly me   to   the moon  "));
    System.out.println("Test 3: \'luffy is still joyboy\' = " + lengthOfLastWord("luffy is still joyboy"));
    System.out.println("Test 4: \'Hello\' = " + lengthOfLastWord("Hello"));
    System.out.println("Test 5: \'a \' = " + lengthOfLastWord("a "));
  }

  public static int lengthOfLastWord(String s){
    int len = 0;
    boolean hit = false;

    for(int i = s.length() - 1; i >= 0 ; i--){
      if(s.charAt(i) == ' ' && hit){
        return len;
      }else if(s.charAt(i) != ' '){
        hit = true;
        len++;
      }else{
        continue;
      }
    }

    return len;
  }
}

File: Python/LengthOfLastWord.py
def lengthOfLastWord(s: str):
  """Determines the character length of the last word in the string

  Args:
      s (str): String being analyzed

  Returns:
      int: length of last word in string
  """
  length = 0
  hit = False
  i = len(s) - 1

  while i >= 0:
    if s[i] == " " and hit == True:
      return length
    elif s[i] != " ":
      hit = True
      length += 1
    i -= 1

  return length

def main():
  print(
    format("%-10s %-30s %10s" %("Test", "String", "Length")) + "\n" +
    "-" * 52 + "\n" +
    format("%-10s %-30s %10d" %("Test 1:", "\'Hello World\'", lengthOfLastWord("Hello World"))) + "\n" +
    format("%-10s %-30s %10d" %("Test 2:", "\'   fly me  to  the moon  \'", lengthOfLastWord("   fly me   to   the moon  "))) + "\n" +
    format("%-10s %-30s %10d" %("Test 3:", "\'luffy is still joyboy\'", lengthOfLastWord("luffy is still joyboy"))) + "\n" +
    format("%-10s %-30s %10d" %("Test 4:", "\'Hello\'", lengthOfLastWord("Hello"))) + "\n" +
    format("%-10s %-30s %10d" %("Test 5:", "\' a\'", lengthOfLastWord(" a"))) + "\n" +
    format("%-10s %-30s %10d" %("Test 6:", "\'a \'", lengthOfLastWord("a ")))
  )

main()